core/field_number.js

```diff
@@ -32,23 +32,40 @@ goog.require('goog.userAgent');
 
 
 /**
+ * Return an appropriate restrictor, depending on whether this FieldNumber
+ * allows decimal or negative numbers.
+ * @param {boolean} decimalAllowed Whether number may have decimal/float component.
+ * @param {boolean} negativeAllowed Whether number may be negative.
+ * @return {!RegExp} Regular expression for this FieldNumber's restrictor.
+ */
+var getNumRestrictor = function(decimalAllowed, negativeAllowed) {
+  var pattern = "[\\d]|[e]"; // Always allow digits and e, from Scratch 2.0.
+  if (decimalAllowed) {
+    pattern += "|[\\.]";
+  }
+  if (negativeAllowed) {
+    pattern += "|[-]";
+  }
+  return new RegExp(pattern);
+};
+
+/**
  * Class for an editable number field.
  * @param {string} text The initial content of the field.
+ * @param {boolean} decimalAllowed Whether number may have decimal/float component.
+ * @param {boolean} negativeAllowed Whether number may be negative.
  * @param {Function=} opt_validator An optional function that is called
  *     to validate any constraints on what the user entered.  Takes the new
  *     text as an argument and returns the accepted text or null to abort
  *     the change.
- * @param {number} precision Precision of the decimal value (negative power of 10).
- * @param {number} min Minimum value of the number.
- * @param {number} max Maximum value of the number.
  * @extends {Blockly.FieldTextInput}
  * @constructor
  */
-Blockly.FieldNumber = function(text, opt_validator, precision, min, max) {
-  this.precision_ = precision;
-  this.min_ = min;
-  this.max_ = max;
-  Blockly.FieldNumber.superClass_.constructor.call(this, text, opt_validator);
+Blockly.FieldNumber = function(text, decimalAllowed, negativeAllowed, opt_validator) {
```

Comment:
You can still use the input restrictions by checking if min && min>0 for '-' and precision && Math.floor(precision) == precision for '.'. To fully support range and precision they would need to be checked by the validator, but starting with the same properties will make it easier to integrate when we add the full support to Blockly.